site/lib/src/components/pages/changelog/changelog_filters_sidebar.dart

```diff
@@ -0,0 +1,331 @@
+                  .text('info'),
+                ]),
+              ],
+            ),
+          ]),
+          ListenableBuilder(
+            listenable: ChangelogFiltersSidebar.filters,
+            builder: (context) {
+              // Ensure we have populated available filters if running on web.
+              if (kIsWeb &&
+                  (ChangelogFiltersSidebar.filters.availableAreas.isEmpty ||
+                      ChangelogFiltersSidebar
+                          .filters
+                          .availableVersions
+                          .isEmpty)) {
+                // Do not populate here to avoid side effects during build.
+                // It should be done in initState.
+              }
+
+              final groups = <String, List<String>>{};
+              for (final version
+                  in ChangelogFiltersSidebar.filters.availableVersions) {
+                final major = version.split('.').first;
+                groups.putIfAbsent(major, () => []).add(version);
+              }
+              for (final group in groups.values) {
+                group.sort(
+                  (left, right) => ChangelogFiltersSidebar.filters
+                      .compareVersions(right, left),
+                );
+              }
+              final sortedMajors = groups.keys.toList()
+                ..sort(
+                  (left, right) => int.parse(right).compareTo(int.parse(left)),
+                );
+
+              return div(classes: 'table-content', [
+                CheckboxFilterGroup<ChangelogTag>(
+                  title: 'Type',
+                  items: ChangelogTag.values
+                      .where((tag) => tag != ChangelogTag.none)
+                      .toList(),
+                  selectedItems: ChangelogFiltersSidebar.filters.selectedTags,
+                  onToggle: (tag, checked) =>
+                      ChangelogFiltersSidebar.filters.setTag(tag, checked),
+                  labelProvider: (tag) => tag.label,
+                  idProvider: (tag) => tag.id,
+                ),
+                CheckboxFilterGroup<String>(
+                  title: 'Area',
+                  items:
+                      ChangelogFiltersSidebar.filters.availableAreas
+                          .where((area) => area != 'Docs')
+                          .toList()
+                        ..sort(ChangelogFiltersSidebar.filters.compareAreas),
+                  selectedItems: ChangelogFiltersSidebar.filters.selectedAreas,
+                  onToggle: (area, checked) =>
+                      ChangelogFiltersSidebar.filters.setArea(area, checked),
+                  labelProvider: (area) => area,
+                  idProvider: (area) => 'area-${slugify(area)}',
+                ),
+                div([
+                  const h4([.text('Version')]),
+                  for (final major in sortedMajors)
+                    CollapsibleFilterGroup<String>(
```

Comment:
These end up with quite a lot of entries. Since I imagine filtering by a specific minor version will be one of the most common ways to filter, we might want to explore ways to not hide them by default. Definitely can be done as part of follow-up improvements though.